src/util.rs

```diff
@@ -62,6 +62,29 @@ where
 {
 }
 
+pub trait PushSimple<P> {
+    fn push_simple(&mut self, other: P);
+}
+
+impl PushSimple<PathBuf> for PathBuf {
+    fn push_simple(&mut self, other: PathBuf) {
```

Comment:
I would suggest leaving out the type parameter and going for
```rust
fn push_simple(&mut self, other: &Path) { ... }
```
as the generality of the type parameter seems unnecessary and `other: PathBuf` forces a temporary allocation as written above.